server/commands.lua

```diff
@@ -1,64 +1,118 @@
-lib.addCommand('givekeys', {
-    help = locale("addcom.givekeys"),
+-----------------------
+----    Imports    ----
+-----------------------
+
+local functions = require 'server.functions'
+
+local removeKeys = functions.removeKeys
+local giveKeys = functions.giveKeys
+
+-----------------------
+----   Commands    ----
+-----------------------
+
+lib.addCommand(locale('addcom.givekeys'), {
+    help = locale('addcom.givekeys_help'),
     params = {
         {
-            name = locale("addcom.givekeys_id"),
-            type = 'number',
-            help = locale("addcom.givekeys_id_help"),
+            name = locale('addcom.givekeys_id'),
+            type = 'playerId',
+            help = locale('addcom.givekeys_id_help'),
+            optional = true
+        },
+        {
+            name = locale('addcom.givekeys_plate'),
+            type = 'string',
+            help = locale('addcom.addkeys_plate_help'),
             optional = true
         },
     },
     restricted = false,
 }, function (source, args)
-    TriggerClientEvent('qb-vehiclekeys:client:GiveKeys', source, args.id)
+    local id = args[locale('addcom.givekeys_id')]
+    if id and not exports.qbx_core:GetPlayer(id) then
+        return exports.qbx_core:Notify(source, locale('notify.player_offline'), 'error')
+    end
+
+    TriggerClientEvent('qb-vehiclekeys:client:GiveKeys', source, id, args[locale('addcom.givekeys_plate')])
 end)
 
-lib.addCommand('addkeys', {
-    help = locale("addcom.addkeys"),
+--- Gets the plate of the vehicle in which the executor is, if args.plate is nil
+---@param source number ID of the player
+---@param plate string? The plate number of the vehicle.
+---@return string?
+local function getPlayersVehiclePlate(source, plate)
+    if not plate then
+        local ped = GetPlayerPed(source)
+        local vehicle = GetVehiclePedIsIn(ped, false)
+
+        if vehicle == 0 then return end
+        plate = GetVehicleNumberPlateText(vehicle)
+    end
+
+    return plate
+end
+
+lib.addCommand(locale('addcom.addkeys'), {
+    help = locale('addcom.addkeys_help'),
     params = {
         {
-            name = 'id',
-            type = 'number',
-            help = locale("addcom.addkeys_id_help"),
+            name = locale('addcom.addkeys_id'),
+            type = 'playerId',
+            help = locale('addcom.addkeys_id_help'),
             optional = true
         },
         {
-            name = 'plate',
+            name = locale('addcom.addkeys_plate'),
             type = 'string',
-            help = locale("addcom.addkeys_plate_help"),
+            help = locale('addcom.addkeys_plate_help'),
             optional = true
         },
     },
     restricted = 'group.admin',
 }, function (source, args)
-    if not args.id or not args.plate then
-        return exports.qbx_core:Notify(source, locale("notify.fpid"))
+    local playerId = args[locale('addcom.addkeys_id')] or source
```

Comment:
Remove the `or source`. That is already handled by the playerId type (by typing `me` instead of a source), and remove the optional part too as this then does not support an optional playerId.